### PR TITLE
upgrade to dotnet 8

### LINKED ISSUE #4: Is this still an active project?
I am interested in what you are doing however I am wondering if this is still an active project before I commit to using it.

### LINKED ISSUE #8: Invalid cast from System.Decimal to 'System.Nullable`
When data is retrieved and a parameter is null, the following error occurs:
Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Invalid cast from 'System.Decimal' to 'System.Nullable`1[[System.Int32, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]'.
      
Example:
 [MagicTable("Test", DbNames.Data)]
 public class TestDTO
 {
     [MagicPrimaryKey("IdTest")]
     public int IdTest { get; set; }

     public int? Age { get; set; }
 }
 
If Age is null it gives that message

### PR DESCRIPTION
**Pull request by yueyinqiu**
I'm back to some blazor stuffs. Really happy to find that you are still here! Although I don't much rely on IndexDb now but I'm glad to contribute to this project.

In this PR I just simply upgraded the project to dotnet 8, since I don't have dotnet 7 installed on my machine hah...

Also upgraded the packages and dexie.

As a side effect of this PR:
- closes #4 (it's a question and it's answered)
- closes #8 (fixed by #10)

And I decide to do those things later:
- replace the event callback style with async functions in JavaScript directly (#2)
- id of the newly added record (#9)
- nullable reference warnings (#15)
- ... (not that familiar with other issues. maybe we could consider them later?)

### CODE REVIEW

File: IndexDb.Example/IndexDb.Example.csproj [hunk 1]
```diff
@@ -1,15 +1,16 @@
-<Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly">
+﻿<Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly">
 
   <PropertyGroup>
-    <TargetFramework>net7.0</TargetFramework>
+    <TargetFramework>net9.0</TargetFramework>
```
Comment (yueyinqiu) [inline IndexDb.Example/IndexDb.Example.csproj:4]: In the example project I turned to dotnet 9 for the latest features. Or maybe we'd like to use dotnet 9 for the library as well?

File: Magic.IndexedDb/wwwroot/magicDB.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import { } from "./dexie.js";
```
Comment (yueyinqiu) [inline Magic.IndexedDb/wwwroot/magicDB.js:1]: Also modified the way we import dexie.

By the way I'm not familiar with JavaScript so is it recommended to `import { }`? Anyway the test project works with this.

(And it does not work if I use `import Dexie from "./dexie.js"`. It says `SyntaxError: The requested module './dexie.js' does not provide an export named 'default'`.)

Comment (yueyinqiu) [reply Magic.IndexedDb/wwwroot/magicDB.js:1]: Oh I solved this by referencing the `.mjs` file rather than `.js`.

**Comment by yueyinqiu** [comment]
Hmm? Why there is a merge from my `async` branch to `8`... I have revent them.

**Review by magiccodingman — APPROVED** [review]

**Comment by magiccodingman** [comment]
Merged. Thanks again!

**Pull request merged**